src/gui/attributetable/qgsdualview.cpp

```diff
@@ -348,6 +348,17 @@ int QgsDualView::filteredFeatureCount()
 
 void QgsDualView::viewWillShowContextMenu( QMenu* menu, const QModelIndex& atIndex )
 {
+  if ( !menu )
+  {
+    return;
+  }
+
+  QgsVectorLayer* vl = mFilterModel->layer();
+  if ( vl && vl->geometryType() != QGis::NoGeometry )
```

Comment:
The check for mapcanvas being a nullptr should probably also be moved here. This way the entry does not even show up instead of doing nothing at a later point.